libs/content/workspace/src/storage_viewer/data.rs

```diff
@@ -0,0 +1,162 @@
+use lb_rs::Uuid;
+use lb_rs::{blocking::Lb, model::file::File};
+use serde::Deserialize;
+use std::collections::HashMap;
+
+#[derive(Debug)]
+pub struct Data {
+    pub current_root: Uuid,
+    pub all_files: HashMap<Uuid, FileRow>,
+    pub folder_sizes: HashMap<Uuid, u64>,
+    pub overall_root: Uuid,
+}
+
+#[derive(PartialEq, Debug, Clone)]
+pub struct Node {
+    pub id: Uuid,
+    pub name: String,
+    pub portion: f32,
+    pub children: Vec<Node>,
+}
+
+#[derive(PartialEq, Debug, Clone)]
+pub struct NodeLayer {
+    pub id: Uuid,
+    pub name: String,
+    pub portion: f32,
+    pub layer: u64,
+}
+
+#[derive(Debug, Deserialize, Clone, Hash, PartialEq, Eq)]
+pub struct FileRow {
+    pub file: File,
+    pub size: u64,
+}
+
+impl Data {
+    pub fn get_filerows(core: Lb) -> Vec<FileRow> {
+        let mut filerows = vec![];
+        let usage = core.get_usage().unwrap().usages;
```

Comment:
This will take an unknown amount of time / could fail. You should visualize and handle both of these situations.